File: codes/csharp/011_operator_spread.cs
using System; // Console, Func<>, Action<>, Action
using System.Linq; // Enumerable
using System.Collections.Generic; // Dictionary<>, List<>, KeyValuePair<>, IEnumerable<>

class Program {
    static void Main(string[] Args) {
        dynamic IsNumeric = (Func<dynamic, bool>)((dynamic Anything) => (Anything is sbyte || Anything is byte || Anything is short || Anything is ushort || Anything is int || Anything is uint || Anything is long || Anything is ulong || Anything is float || Anything is double || Anything is decimal));

        string JsonStringify(dynamic Anything, bool Pretty = false, string Indent = "    ") {
            int IndentLevel = 0;
            string JsonStringifyInner(dynamic AnythingInner, string IndentInner) {
                if (AnythingInner == null) return "null";
                if (AnythingInner is string) return "\"" + (string)AnythingInner + "\"";
                if ((IsNumeric(AnythingInner) == true) || AnythingInner is bool) return AnythingInner.ToString().Replace(",", ".");
                if (AnythingInner is List<dynamic>) {
                    IndentLevel += 1;
                    string Result = ((Pretty == true) ? ("[" + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : "[");
                    int ArrayItemIndex = 0;
                    foreach (dynamic ArrayItem in (List<dynamic>)AnythingInner) {
                        Result += JsonStringifyInner(ArrayItem, IndentInner);
                        if ((ArrayItemIndex + 1) != ((List<dynamic>)AnythingInner).Count) Result += ((Pretty == true) ? ("," + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : ", ");
                        ArrayItemIndex += 1;
                    }
                    IndentLevel -= 1;
                    Result += ((Pretty == true) ? (Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) + "]") : "]");
                    return Result;
                }
                if (AnythingInner is Dictionary<string, dynamic>) {
                    IndentLevel += 1;
                    string Result = ((Pretty == true) ? ("{" + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : "{");
                    int ObjectIterationIndex = 0;
                    foreach (KeyValuePair<string, dynamic> ObjectEntry in (Dictionary<string, dynamic>)AnythingInner) {
                        string ObjectKey = ObjectEntry.Key;
                        dynamic ObjectValue = ObjectEntry.Value;
                        Result += "\"" + ObjectKey + "\": " + JsonStringifyInner(ObjectValue, IndentInner);
                        if ((ObjectIterationIndex + 1) != ((Dictionary<string, dynamic>)AnythingInner).Count) Result += ((Pretty == true) ? ("," + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : ", ");
                        ObjectIterationIndex += 1;
                    }
                    IndentLevel -= 1;
                    Result += ((Pretty == true) ? (Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) + "}") : "}");
                    return Result;
                }
                return "null";
            };
            return JsonStringifyInner(Anything, Indent);
        }

        Dictionary<string, dynamic> SpreadObject(params dynamic[] Arguments) {
            dynamic NewObject = new Dictionary<string, dynamic>();
            int ArgumentIndex = 0;
            foreach (dynamic Argument in Arguments) {
                if (Argument is Dictionary<string, dynamic>) {
                    int ObjectIterationIndex = 0;
                    foreach (KeyValuePair<string, dynamic> ObjectEntry in (Dictionary<string, dynamic>)Argument) {
                        string ObjectKey = ObjectEntry.Key;
                        dynamic ObjectValue = ObjectEntry.Value;
                        NewObject[Convert.ToString(ObjectKey)] = ObjectValue;
                        ObjectIterationIndex += 1;
                    }
                    continue;
                }
                if (Argument is List<dynamic>) {
                    int ArrayItemIndex = 0;
                    foreach (dynamic ArrayItem in (List<dynamic>)Argument) {
                        NewObject.Add(Convert.ToString(ArrayItemIndex), ArrayItem);
                        ArrayItemIndex += 1;
                    }
                    continue;
                }
                ArgumentIndex += 1;
            }
            return NewObject;
        }

        List<dynamic> SpreadArray(params dynamic[] Arguments) {
            dynamic NewArray = new List<dynamic>();
            int ArgumentIndex = 0;
            foreach (dynamic Argument in Arguments) {
                if (Argument is Dictionary<string, dynamic>) {
                    if (Argument.Count == 1) {
                        int ObjectIterationIndex = 0;
                        foreach (KeyValuePair<string, dynamic> ObjectEntry in (Dictionary<string, dynamic>)Argument) {
                            string ObjectKey = ObjectEntry.Key;
                            dynamic ObjectValue = ObjectEntry.Value;
                            NewArray.Add(ObjectValue);
                            ObjectIterationIndex += 1;
                        }
                        continue;
                    }
                    NewArray.Add(Argument);
                    continue;
                }
                if (Argument is List<dynamic>) {
                    int ArrayItemIndex = 0;
                    foreach (dynamic ArrayItem in (List<dynamic>)Argument) {
                        NewArray.Add(ArrayItem);
                        ArrayItemIndex += 1;
                    }
                    continue;
                }
                ArgumentIndex += 1;
            }
            return NewArray;
        }

        Console.WriteLine("\n// JavaScript-like Spread Syntax (...) in C#");

        // There's no JavaScript-like Spread Syntax (...) in C#.
        // But, we can create our own function to mimic it in C#.

        dynamic Fruits = new List<dynamic>() {"Mango", "Melon", "Banana"};
        Console.WriteLine($"Fruits: {JsonStringify(Fruits)}");

        dynamic Vegetables = new List<dynamic>() {"Carrot", "Tomato"};
        Console.WriteLine($"Vegetables: {JsonStringify(Vegetables)}");

        dynamic CountryCapitalsInAsia = new Dictionary<string, dynamic>() {
            {"Thailand", "Bangkok"},
            {"China", "Beijing"},
            {"Japan", "Tokyo"}
        };
        Console.WriteLine($"CountryCapitalsInAsia: {JsonStringify(CountryCapitalsInAsia, Pretty: true)}");

        dynamic CountryCapitalsInEurope = new Dictionary<string, dynamic>() {
            {"France", "Paris"},
            {"England", "London"}
        };
        Console.WriteLine($"CountryCapitalsInEurope: {JsonStringify(CountryCapitalsInEurope, Pretty: true)}");

        Console.WriteLine("\n// [...array1, ...array2]:\n");

        dynamic Combination1 = SpreadArray(Fruits, Vegetables);
        Console.WriteLine($"Combination1: {JsonStringify(Combination1, Pretty: true)}");
        // Combination1: [
        //     "Mango",
        //     "Melon",
        //     "Banana",
        //     "Carrot",
        //     "Tomato"
        // ]

        dynamic Combination2 = SpreadArray(Fruits, new List<dynamic>() {"Cucumber", "Cabbage"});
        Console.WriteLine($"Combination2: {JsonStringify(Combination2, Pretty: true)}");
        // Combination2: [
        //     "Mango",
        //     "Melon",
        //     "Banana",
        //     "Cucumber",
        //     "Cabbage"
        // ]

        Console.WriteLine("\n// { ...object1, ...object2 }:\n");

        dynamic Combination3 = SpreadObject(CountryCapitalsInAsia, CountryCapitalsInEurope);
        Console.WriteLine($"Combination3: {JsonStringify(Combination3, Pretty: true)}");
        // Combination3: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "France": "Paris",
        //     "England": "London"
        // }

        dynamic Combination4 = SpreadObject(CountryCapitalsInAsia, new Dictionary<string, dynamic>() {{"Germany", "Berlin"}, {"Italy", "Rome"}});
        Console.WriteLine($"Combination4: {JsonStringify(Combination4, Pretty: true)}");
        // Combination4: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "Germany": "Berlin",
        //     "Italy": "Rome"
        // }

        Console.WriteLine("\n// [...array1, array2] || [...array1, newArrayItem1, newArrayItem2]:\n");

        dynamic Combination5 = SpreadArray(Fruits, new Dictionary<string, dynamic>() {{"Vegetables", Vegetables}});
        Console.WriteLine($"Combination5: {JsonStringify(Combination5, Pretty: true)}");
        // Combination5: [
        //     "Mango",
        //     "Melon",
        //     "Banana",
        //     [
        //         "Carrot",
        //         "Tomato"
        //     ]
        // ]

        dynamic Combination6 = SpreadArray(Fruits, new Dictionary<string, dynamic>() {{"Vegetables", new List<dynamic>() {"Cucumber", "Cabbage"}}});
        Console.WriteLine($"Combination6: {JsonStringify(Combination6, Pretty: true)}");
        // Combination6: [
        //     "Mango",
        //     "Melon",
        //     "Banana",
        //     [
        //         "Cucumber",
        //         "Cabbage"
        //     ]
        // ]

        Console.WriteLine("\n// [...array1, object1] || [...array1, newArrayItem1, newArrayItem2]:\n");

        dynamic Combination7 = SpreadArray(Fruits, new Dictionary<string, dynamic>() {{"CountryCapitalsInAsia", CountryCapitalsInAsia}});
        Console.WriteLine($"Combination7: {JsonStringify(Combination7, Pretty: true)}");
        // Combination7: [
        //     "Mango",
        //     "Melon",
        //     "Banana",
        //     {
        //         "Thailand": "Bangkok",
        //         "China": "Beijing",
        //         "Japan": "Tokyo"
        //     }
        // ]

        dynamic Combination8 = SpreadArray(Fruits, new Dictionary<string, dynamic>() {{"CountryCapitalsInEurope", new Dictionary<string, dynamic>() {{"Germany", "Berlin"}, {"Italy", "Rome"}}}});
        Console.WriteLine($"Combination8: {JsonStringify(Combination8, Pretty: true)}");
        // Combination8: [
        //     "Mango",
        //     "Melon",
        //     "Banana",
        //     {
        //         "Germany": "Berlin",
        //         "Italy": "Rome"
        //     }
        // ]

        Console.WriteLine("\n// { ...object1, object2 } || { ...object1, objectKey: objectValue }:\n");

        dynamic Combination9 = SpreadObject(CountryCapitalsInAsia, new Dictionary<string, dynamic>() {{"CountryCapitalsInEurope", CountryCapitalsInEurope}});
        Console.WriteLine($"Combination9: {JsonStringify(Combination9, Pretty: true)}");
        // Combination9: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "CountryCapitalsInEurope": {
        //         "France": "Paris",
        //         "England": "London"
        //     }
        // }

        dynamic Combination10 = SpreadObject(CountryCapitalsInAsia, new Dictionary<string, dynamic>() {{"CountryCapitalsInEurope", new Dictionary<string, dynamic>() {{"Germany", "Berlin"}, {"Italy", "Rome"}}}});
        Console.WriteLine($"Combination10: {JsonStringify(Combination10, Pretty: true)}");
        // Combination10: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "CountryCapitalsInEurope": {
        //         "Germany": "Berlin",
        //         "Italy": "Rome"
        //     }
        // }

        Console.WriteLine("\n// { ...object1, array2 } || { ...object1, objectKey: objectValue }:\n");

        dynamic Combination11 = SpreadObject(CountryCapitalsInAsia, new Dictionary<string, dynamic>() {{"Vegetables", Vegetables}});
        Console.WriteLine($"Combination11: {JsonStringify(Combination11, Pretty: true)}");
        // Combination11: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "Vegetables": [
        //         "Carrot",
        //         "Tomato"
        //     ]
        // }

        dynamic Combination12 = SpreadObject(CountryCapitalsInAsia, new Dictionary<string, dynamic>() {{"Vegetables", new List<dynamic>() {"Cucumber", "Cabbage"}}});
        Console.WriteLine($"Combination12: {JsonStringify(Combination12, Pretty: true)}");
        // Combination12: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "Vegetables": [
        //         "Cucumber",
        //         "Cabbage"
        //     ]
        // }

        Console.WriteLine("\n// { ...object1, ...array2 }:\n");

        dynamic Combination13 = SpreadObject(CountryCapitalsInAsia, Vegetables);
        Console.WriteLine($"Combination13: {JsonStringify(Combination13, Pretty: true)}");
        // Combination13: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "0": "Carrot",
        //     "1": "Tomato"
        // }

        dynamic Combination14 = SpreadObject(CountryCapitalsInAsia, new List<dynamic>() {"Cucumber", "Cabbage"});
        Console.WriteLine($"Combination14: {JsonStringify(Combination14, Pretty: true)}");
        // Combination14: {
        //     "Thailand": "Bangkok",
        //     "China": "Beijing",
        //     "Japan": "Tokyo",
        //     "0": "Cucumber",
        //     "1": "Cabbage"
        // }

        // Console.WriteLine("\n// [...array1, ...object1]: // this combination throw an error in JavaScript\n");

        // this combination throw an error in JavaScript
        // dynamic CombinationErrorInJavascript1 = SpreadArray(Fruits, CountryCapitalsInAsia);
        // Console.WriteLine($"CombinationErrorInJavascript1: {JsonStringify(CombinationErrorInJavascript1, Pretty: true)}");

        // this combination throw an error in JavaScript
        // dynamic CombinationErrorInJavascript2 = SpreadArray(Fruits, new Dictionary<string, dynamic>() {{"Germany", "Berlin"}, {"Italy", "Rome"}});
        // Console.WriteLine($"CombinationErrorInJavascript2: {JsonStringify(CombinationErrorInJavascript2, Pretty: true)}");
    }
}

File: codes/swift/011_operator_spread.swift
import Foundation

typealias MyObject = [String: Any?]
typealias MyArray = [Any?]

func jsonStringify(_ anything: Any? = nil, pretty: Bool = false, indent: String = "    ") -> String {
    var indentLevel = 0
    func jsonStringifyInner(_ anythingInner: Any?, _ indentInner: String) -> String {
        guard let anythingInner = anythingInner else {
            return "null"
        }
        if let anythingInner = anythingInner as? String {
            return "\"\(anythingInner)\""
        }
        if let anythingInner = anythingInner as? Bool {
            return "\(anythingInner)"
        }
        if let anythingInner = anythingInner as? NSNumber {
            return "\(anythingInner)"
        }
        if let anythingInner = anythingInner as? MyArray {
            if (anythingInner.count == 0) {
                return "[]"
            }
            indentLevel += 1
            var result = ((pretty == true) ? "[\n\(String(repeating: indentInner, count: indentLevel))" : "[")
            for (arrayItemIndex, arrayItem) in anythingInner.enumerated() {
                result += jsonStringifyInner(arrayItem, indentInner)
                if ((arrayItemIndex + 1) != anythingInner.count) {
                    result += ((pretty == true) ? ",\n\(String(repeating: indentInner, count: indentLevel))" : ", ")
                }
            }
            indentLevel -= 1
            result += ((pretty == true) ? "\n\(String(repeating: indentInner, count: indentLevel))]" : "]")
            return result
        }
        if let anythingInner = anythingInner as? MyObject {
            if (anythingInner.count == 0) {
                return "{}"
            }
            indentLevel += 1
            var result = ((pretty == true) ? "{\n\(String(repeating: indentInner, count: indentLevel))" : "{")
            for (objectEntryIndex, (objectKey, objectValue)) in anythingInner.enumerated() {
                result += "\"\(objectKey)\": \(jsonStringifyInner(objectValue, indentInner))"
                if ((objectEntryIndex + 1) != anythingInner.count) {
                    result += ((pretty == true) ? ",\n\(String(repeating: indentInner, count: indentLevel))" : ", ")
                }
            }
            indentLevel -= 1
            result += ((pretty == true) ? "\n\(String(repeating: indentInner, count: indentLevel))}" : "}")
            return result
        }
        return "null"
    }
    return jsonStringifyInner(anything, indent)
}

// There's no JavaScript-like Spread Syntax (...) in Swift.
// But, we can create our own function to mimic it in Swift.

func spreadObject(_ arguments: Any?...) -> MyObject {
    var newObject = MyObject()
    for (_, argument) in arguments.enumerated() {
        if (argument is MyObject) {
            guard let argument = argument as? MyObject else {
                continue
            }
            for (objectKey, objectValue) in argument {
                newObject[objectKey] = objectValue
            }
            continue
        }
        if (argument is MyArray) {
            guard let argument = argument as? MyArray else {
                continue
            }
            for (arrayItemIndex, arrayItem) in argument.enumerated() {
                newObject[String(arrayItemIndex)] = arrayItem
            }
            continue
        }
    }
    return newObject
}

func spreadArray(_ arguments: Any?...) -> MyArray {
    var newArray = MyArray()
    for (_, argument) in arguments.enumerated() {
        if (argument is MyObject) {
            guard let argument = argument as? MyObject else {
                continue
            }
            if (argument.count == 1) {
                for (_, objectValue) in argument {
                    newArray.append(objectValue)
                }
                continue
            }
            newArray.append(argument)
            continue
        }
        if (argument is MyArray) {
            guard let argument = argument as? MyArray else {
                continue
            }
            for (_, arrayItem) in argument.enumerated() {
                newArray.append(arrayItem)
            }
            continue
        }
    }
    return newArray
}

print("\n// JavaScript-like Spread Syntax (...) in Swift")

let fruits: MyArray = ["Mango", "Melon", "Banana"]
print("fruits: \(jsonStringify(fruits))")

let vegetables: MyArray = ["Carrot", "Tomato"]
print("vegetables: \(jsonStringify(vegetables))")

var countryCapitalsInAsia: MyObject = [
    "Thailand": "Bangkok",
    "China": "Beijing",
    "Japan": "Tokyo"
]
print("countryCapitalsInAsia: \(jsonStringify(countryCapitalsInAsia, pretty: true))")

var countryCapitalsInEurope: MyObject = [
    "France": "Paris",
    "England": "London"
]
print("countryCapitalsInEurope: \(jsonStringify(countryCapitalsInEurope, pretty: true))")

print("\n// [...array1, ...array2]:\n")

let combination1 = spreadArray(fruits, vegetables)
print("combination1: \(jsonStringify(combination1, pretty: true))")
// combination1: [
//     "Mango",
//     "Melon",
//     "Banana",
//     "Carrot",
//     "Tomato"
// ]

let combination2 = spreadArray(fruits, ["Cucumber", "Cabbage"] as MyArray)
print("combination2: \(jsonStringify(combination2, pretty: true))")
// combination2: [
//     "Mango",
//     "Melon",
//     "Banana",
//     "Cucumber",
//     "Cabbage"
// ]

print("\n// { ...object1, ...object2 }:\n")

let combination3 = spreadObject(countryCapitalsInAsia, countryCapitalsInEurope)
print("combination3: \(jsonStringify(combination3, pretty: true))")
// combination3: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "France": "Paris",
//     "England": "London"
// }

let combination4 = spreadObject(countryCapitalsInAsia, ["Germany": "Berlin", "Italy": "Rome"] as MyObject)
print("combination4: \(jsonStringify(combination4, pretty: true))")
// combination4: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "Germany": "Berlin",
//     "Italy": "Rome"
// }

print("\n// [...array1, array2] || [...array1, newArrayItem1, newArrayItem2]:\n")

let combination5 = spreadArray(fruits, ["vegetables": vegetables] as MyObject)
print("combination5: \(jsonStringify(combination5, pretty: true))")
// combination5: [
//     "Mango",
//     "Melon",
//     "Banana",
//     [
//         "Carrot",
//         "Tomato"
//     ]
// ]

let combination6 = spreadArray(fruits, ["vegetables": ["Cucumber", "Cabbage"] as MyArray] as MyObject)
print("combination6: \(jsonStringify(combination6, pretty: true))")
// combination6: [
//     "Mango",
//     "Melon",
//     "Banana",
//     [
//         "Cucumber",
//         "Cabbage"
//     ]
// ]

print("\n// [...array1, object1] || [...array1, newArrayItem1, newArrayItem2]:\n")

let combination7 = spreadArray(fruits, ["countryCapitalsInAsia": countryCapitalsInAsia] as MyObject)
print("combination7: \(jsonStringify(combination7, pretty: true))")
// combination7: [
//     "Mango",
//     "Melon",
//     "Banana",
//     {
//         "Thailand": "Bangkok",
//         "China": "Beijing",
//         "Japan": "Tokyo"
//     }
// ]

let combination8 = spreadArray(fruits, ["countryCapitalsInEurope": ["Germany": "Berlin", "Italy": "Rome"] as MyObject] as MyObject)
print("combination8: \(jsonStringify(combination8, pretty: true))")
// combination8: [
//     "Mango",
//     "Melon",
//     "Banana",
//     {
//         "Germany": "Berlin",
//         "Italy": "Rome"
//     }
// ]

print("\n// { ...object1, object2 } || { ...object1, objectKey: objectValue }:\n")

let combination9 = spreadObject(countryCapitalsInAsia, ["countryCapitalsInEurope": countryCapitalsInEurope] as MyObject)
print("combination9: \(jsonStringify(combination9, pretty: true))")
// combination9: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "countryCapitalsInEurope": {
//         "France": "Paris",
//         "England": "London"
//     }
// }

let combination10 = spreadObject(countryCapitalsInAsia, ["countryCapitalsInEurope": ["Germany": "Berlin", "Italy": "Rome"] as MyObject] as MyObject)
print("combination10: \(jsonStringify(combination10, pretty: true))")
// combination10: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "countryCapitalsInEurope": {
//         "Germany": "Berlin",
//         "Italy": "Rome"
//     }
// }

print("\n// { ...object1, array2 } || { ...object1, objectKey: objectValue }:\n")

let combination11 = spreadObject(countryCapitalsInAsia, ["vegetables": vegetables] as MyObject)
print("combination11: \(jsonStringify(combination11, pretty: true))")
// combination11: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "vegetables": [
//         "Carrot",
//         "Tomato"
//     ]
// }

let combination12 = spreadObject(countryCapitalsInAsia, ["vegetables": ["Cucumber", "Cabbage"] as MyArray] as MyObject)
print("combination12: \(jsonStringify(combination12, pretty: true))")
// combination12: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "vegetables": [
//         "Cucumber",
//         "Cabbage"
//     ]
// }

print("\n// { ...object1, ...array2 }:\n")

let combination13 = spreadObject(countryCapitalsInAsia, vegetables)
print("combination13: \(jsonStringify(combination13, pretty: true))")
// combination13: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "0": "Carrot",
//     "1": "Tomato"
// }

let combination14 = spreadObject(countryCapitalsInAsia, ["Cucumber", "Cabbage"] as MyArray)
print("combination14: \(jsonStringify(combination14, pretty: true))")
// combination14: {
//     "Thailand": "Bangkok",
//     "China": "Beijing",
//     "Japan": "Tokyo",
//     "0": "Cucumber",
//     "1": "Cabbage"
// }

// print("\n// [...array1, ...object1]: // this combination throw an error in JavaScript\n")

// this combination throw an error in JavaScript
// let combinationErrorInJavascript1 = spreadArray(fruits, countryCapitalsInAsia)
// print("combinationErrorInJavascript1: \(jsonStringify(combinationErrorInJavascript1, pretty: true))")

// this combination throw an error in JavaScript
// let combinationErrorInJavascript2 = spreadArray(fruits, ["Germany": "Berlin", "Italy": "Rome"] as MyObject)
// print("combinationErrorInJavascript2: \(jsonStringify(combinationErrorInJavascript2, pretty: true))")
